### PR TITLE
Bu 2 d migrate 60 712

### PR DESCRIPTION
**Pull request by JohannaOtt**
SVC-1467

### CODE REVIEW

**Review by florianesser — COMMENTED** [review]
Left a couple of comments in the diffs. Also, in the commit message of `feat: add buffer value for Burg`, please format the link to the hc task as a Markdown link.

File: annex-2-3/mappings7.1/Buildings/2D/3A2INSPIRE_BU2D.halex.alignment.xml [hunk 1]
```diff
@@ -2153,7 +2153,15 @@ _target{&#13;
         <parameter value="54120340-6427-4f58-a329-cd48b2b4fc8a" name="lookupTableID"/>
         <parameter value="null" name="notClassifiedAction"/>
     </cell>
-    <cell relation="eu.esdihumboldt.hale.align.assign" id="C9fbf711b-1294-4e75-9fcf-c187e1d85801" priority="normal">
+    <cell relation="eu.esdihumboldt.hale.align.assign.bound" id="C9fbf711b-1294-4e75-9fcf-c187e1d85801" priority="normal">
+        <source name="anchor">
+            <property>
+                <type name="AX_GebaeudeType" ns="http://www.adv-online.de/namespaces/adv/gid/7.1"/>
+                <child name="gebaeudefunktion" ns="http://www.adv-online.de/namespaces/adv/gid/7.1">
+                    <condition lang="ECQL">value &lt;&gt; 9998</condition>
```
[the remark this answers is not in the document]
Comment (JohannaOtt) [reply annex-2-3/mappings7.1/Buildings/2D/3A2INSPIRE_BU2D.halex.alignment.xml:2161]: > doesn't this result in instances of currentUse that have currentUse.href filled but percentage unfilled and without nilReason?

It should not as there is no classification defined for 9998 and therefore, there should also be no currentUse value in these cases.


> A quick check with test data showed more currentUse instances (4405) than nilReasons (3255). Any idea what's going on here?

That could partly be caused by the issue you found with the target side contexts not having a value for percentage at all. Another reason most probably is that the currentUse value is filled from various attributes 
grafik
but for those, no nilReasons had been provided. Fixed that now.

**Comment by florianesser** [comment]
@JohannaOtt Thanks for the update. I retested it with the same data and while the `percentage` elements now appear to have `nilReasons` consistently, I now found objects that have an empty `currentUse` element, e.g. like the one shown below. This fails the schema validation in the INSPIRE Validator.

```
<bu-core2d:Building gml:id="Building_DEHER7LV00006M7J">
      <gml:identifier codeSpace="http://inspire.ec.europa.eu/ids">https://registry.gdi-de.org/id/de.____/Building_DEHER7LV00006M7J</gml:identifier>
      <bu-base:beginLifespanVersion>2012-09-05T11:27:12Z</bu-base:beginLifespanVersion>
      <bu-base:conditionOfConstruction xlink:href="http://inspire.ec.europa.eu/codelist/ConditionOfConstructionValue/functional"></bu-base:conditionOfConstruction>
      <bu-base:inspireId>
        <base:Identifier>
          <base:localId>Building_DEHER7LV00006M7J</base:localId>
          <base:namespace>https://registry.gdi-de.org/id/de.____/</base:namespace>
        </base:Identifier>
      </bu-base:inspireId>
      <bu-base:buildingNature xlink:href="https://registry.gdi-de.org/codelist/de.adv-online.inspire/BuildingNatureValue/mast"></bu-base:buildingNature>
      <bu-base:currentUse>
        <bu-base:CurrentUse>
          <bu-base:currentUse/>
          <bu-base:percentage nilReason="other:unpopulated" xsi:nil="true"></bu-base:percentage>
        </bu-base:CurrentUse>
      </bu-base:currentUse>
      <bu-core2d:geometry2D>
        <bu-base:BuildingGeometry2D>
          <bu-base:geometry>
            <gml:Point srsName="http://www.opengis.net/def/crs/EPSG/0/4258" srsDimension="2">
              <gml:pos>50.144809833200746 8.346783235438815</gml:pos>
            </gml:Point>
          </bu-base:geometry>
          <bu-base:referenceGeometry>true</bu-base:referenceGeometry>
          <bu-base:horizontalGeometryReference xlink:href="http://inspire.ec.europa.eu/codelist/HorizontalGeometryReferenceValue/footPrint"></bu-base:horizontalGeometryReference>
          <bu-base:horizontalGeometryEstimatedAccuracy uom="m" nilReason="other:unpopulated" xsi:nil="true"></bu-base:horizontalGeometryEstimatedAccuracy>
        </bu-base:BuildingGeometry2D>
      </bu-core2d:geometry2D>
    </bu-core2d:Building>
```

**Comment by JohannaOtt** [comment]
@florianesser Which dataset did you use for testing here? I think that would make it a lot easier to fix the issue.

**Comment by JohannaOtt** [comment]
> I now found objects that have an empty currentUse element, e.g. like the one shown below

Mapping table is missing a value - in clarification [here](https://haleconnect.com/#/transformation/org/765/f228a445-c133-4a61-a6fa-04e787c69a9c/tasks/co4i)

**Comment by JohannaOtt** [comment]
> I now found objects that have an empty currentUse element, e.g. like the one shown below

Added the value to the mapping table now. No missing currentUse values are appearing in the data validation any longer when using the example data used for challenging.

**Review by florianesser — APPROVED** [review]
Thanks for the updates, LGTM now!

**Pull request merged**